management/src/main/java/nearsoft/skytouch/management/controller/ProductController.java

```diff
@@ -29,22 +32,38 @@ private String newProductView(Map<String, Object> model) {
     }
 
     @RequestMapping(value = "/products", method = RequestMethod.POST)
-    private String submit(@ModelAttribute("product") Product product,
+    private String submit(@Valid @ModelAttribute("product") Product product,
                           BindingResult bindingResult, Map<String, Object> model) {
         if (bindingResult.hasErrors()) {
-            model.put("errors", bindingResult.getAllErrors());
-            return "errors/error";
+            return "products/newProduct";
         }
-        productService.storeProduct(product);
-        model.put("product", new Product());
-        return "redirect:/view/product";
+        try {
+            productService.storeProduct(product);
+            model.put("product", new Product());
+            return "redirect:/view/product";
+        } catch (AmqpException e) {
+            model.put("errors", "Failed to connect with the server try again");
+            return "products/newProduct";
+        } catch (IOException e) {
+            model.put("errors", "Failed to parse the result");
+            return "products/newProduct";
+        }
+
     }
 
     @RequestMapping(value = "/products", method = RequestMethod.GET)
     private String listProducts(Map<String, Object> model) {
         List<Product> products;
-        products = productService.retrieveProducts();
-        model.put("products", products);
-        return "products/listProducts";
+        try {
+            products = productService.retrieveProducts();
+            model.put("products", products);
```

Comment:
Why you are not using ExceptionHandler?

Reply:
This is an old version, i've not pushed the newest yet